mscore/seq.cpp

```diff
@@ -1314,19 +1314,23 @@ void Seq::stopNotes(int channel, bool realTime)
                   putEvent(event);
             else
                   sendEvent(event);
-      };
+            };
+      auto sendAllNotesOff = [send](int channel) {
+            for (int note = 0; note < 128; note++)
+                  send(NPlayEvent(ME_NOTEOFF, channel, note, 0));
+            };
       // Stop notes in all channels
       if (channel == -1) {
-            for(int ch = 0; ch < int(cs->midiMapping().size()); ch++) {
+            for (int ch = 0; ch < int(cs->midiMapping().size()); ch++) {
                   send(NPlayEvent(ME_CONTROLLER, ch, CTRL_SUSTAIN, 0));
-                  send(NPlayEvent(ME_CONTROLLER, ch, CTRL_ALL_NOTES_OFF, 0));
```

Comment:
I'm not sure we need to remove sending `CTRL_ALL_NOTES_OFF` event. For those hardware/software that recognise that type of event, it still makes sense.

Reply:
While I had the 1.0 specification downloaded, I checked the All Notes Off section, to see if there was anything about All Notes Off being optional/deprecated, and it says this:

> While this message is useful for some applications, there is no requirement that a receiver recognize it. Since recognition of All Notes Off is not required of the receiver, all notes should first be turned off by transmitting individual Note-Off messages prior to sending an All Notes Off.

So you're right, that message should still be sent. I will get this changed later tonight.

Reply:
Thank you, sir! Glad you came back to deal with it :)

Reply:
Sir, "all notes should first be turned off by transmitting individual Note-Off messages prior to sending an All Notes Off", but you implemented this vice versa :)
Could you please place `turnAllNotesOff` call before the call of `send(NPlayEvent(ME_CONTROLLER, ch, CTRL_ALL_NOTES_OFF, 0));`?

Reply:
@ChrootDoot ^